packages/toolpad-components/src/Form.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import { Container as MUIContainer, ContainerProps, Button } from '@mui/material';
+import { createComponent } from '@mui/toolpad-core';
+import { Formik, useFormikContext } from 'formik';
+import { SX_PROP_HELPER_TEXT } from './constants';
+
+type FormValue = Record<string, string>;
+
+interface Props extends ContainerProps {
+  onSubmit: (values: any) => void;
+  submitLabel: string;
+  value: FormValue;
+  onChange: (value: FormValue) => void;
+}
+
+// Because: https://github.com/jaredpalmer/formik/issues/271
+function ChangeHandler({ onChange }: any) {
+  const { values } = useFormikContext();
+
+  React.useEffect(() => {
+    onChange(values);
+  }, [values, onChange]);
+
+  return null;
+}
+
+function Form({ children, onSubmit, onChange, submitLabel, sx, defaultValue, ...rest }: Props) {
+  return (
+    <MUIContainer disableGutters sx={sx} {...rest}>
+      <Formik
+        initialValues={defaultValue}
+        onSubmit={(values) => {
+          onSubmit(values);
+        }}
+      >
+        {(props) => (
+          <form onSubmit={props.handleSubmit}>
+            <MUIContainer />
+            <ChangeHandler onChange={onChange} />
+            {children}
+
+            <MUIContainer
+              disableGutters
+              sx={{
+                borderTop: 'solid 1px',
+                display: 'flex',
+                justifyContent: 'end',
+                width: '100%',
+                mt: 1,
+                pt: 1,
+              }}
+            >
+              <Button type="submit" variant="contained">
+                {submitLabel}
+              </Button>
+            </MUIContainer>
+          </form>
+        )}
+      </Formik>
+    </MUIContainer>
+  );
+}
+
+export default createComponent(Form, {
+  argTypes: {
+    children: {
```

Comment:
I have a fix for the ugly yellow message that shows in the properties panel in my `Add List component` PR, I should probably extract it from it... Then we wouldn't need `visible: false` (I guess that's why you added it, to hide that control warning?).

Reply:
Mmm I already forgot, but it might be that 😅